src/vulkan-renderer/application.cpp

```diff
@@ -571,17 +531,38 @@ void Application::update_imgui_overlay() {
     m_imgui_overlay->update();
 }
 
-void Application::process_mouse_input() {
-    const auto cursor_pos_delta = m_input_data->calculate_cursor_position_delta();
+void Application::process_input() {
+    const auto cursor_pos_delta = m_input->kbm_data().calculate_cursor_position_delta();
 
-    if (m_camera->type() == CameraType::LOOK_AT && m_input_data->is_mouse_button_pressed(GLFW_MOUSE_BUTTON_LEFT)) {
+    auto deadzone_lambda = [](const float state) {
+        if (state > -0.2f && state < 0.2f) {
+            return 0.f;
+        }
+        return state;
+    };
+
+    if (m_camera->type() == CameraType::LOOK_AT &&
+        m_input->kbm_data().is_mouse_button_pressed(GLFW_MOUSE_BUTTON_LEFT)) {
         m_camera->rotate(static_cast<float>(cursor_pos_delta[0]), -static_cast<float>(cursor_pos_delta[1]));
     }
-
-    m_camera->set_movement_state(CameraMovement::FORWARD, m_input_data->is_key_pressed(GLFW_KEY_W));
-    m_camera->set_movement_state(CameraMovement::LEFT, m_input_data->is_key_pressed(GLFW_KEY_A));
-    m_camera->set_movement_state(CameraMovement::BACKWARD, m_input_data->is_key_pressed(GLFW_KEY_S));
-    m_camera->set_movement_state(CameraMovement::RIGHT, m_input_data->is_key_pressed(GLFW_KEY_D));
+    if (m_camera->type() == CameraType::LOOK_AT) {
```

Comment:
This works for now, though in future the `Input` class should abstract away whether the input event came from kb&m or controller, with a mapping system (so both a controller and mouse could be mapped to a "vertical" and "horizontal" axis that the application can query from `m_input`).

Reply:
Yeah in the future I want to refactor the camera code anyways (see https://github.com/inexorgame/vulkan-renderer/pull/474)